subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -154,43 +154,39 @@ int lwm2m_open_socket(struct lwm2m_ctx *client_ctx)
 
 int lwm2m_close_socket(struct lwm2m_ctx *client_ctx)
 {
-	int ret = 0;
-
 	if (client_ctx->sock_fd >= 0) {
-		ret = zsock_close(client_ctx->sock_fd);
+		int ret = zsock_close(client_ctx->sock_fd);
```

Comment:
I believe per the coding standards, this variable definition should remain at the top of the function.

Reply:
Do you have a source for that rule?

Checkpatch only complains that there should be an empty line after introduction of variable.

Reply:
AFAIC it wasn't the strong rule dictated by our CS (AFAIR it's just a heritage of C89). Anyways, even said that the requirement is to declare the variables at the top of the block (scope), not necessarily the top of the function. So I see nothing wrong here.

Reply:
yes, there are many inconsistencies. I just mention it here because it was consciously changed here and it's not clear to me the value of making this change.
A few reasons it's nice to follow this "rule":
1. It's easy to account for stack size requirements of the function.
2. If using `goto` in a function, definitions need to appear before the goto.

Reply:
I see. I think that is a good guideline.

The reason for my refactoring was to get rid of this style that I do not like:
```c
in func()
{
    int ret = 0;

    if (...) {
        /* do stuff */
        ret = -1;
    }
    return ret;
}
```

That style is bad for two reasons:
1. I drops the compiler's ability to warn about unused variable, because you initialize it with zero. This might leave paths that you did not intend to happen and you don't know if they can happen.
2. It gives no hint that last return is the OK path.

So I tend to refactor those by either removing the zero initialization, so I allows compiler or static analyzer to do its work and warn if there are paths where it is uninitialized. Or I limit the scope and use `return 0;` to indicate that all is OK.

Reply:
Thanks, appreciate the points here.